code/datums/components/hattable.dm

```diff
@@ -33,19 +33,21 @@ TYPEINFO(/datum/component/hattable) // Take a walk through my TWISTED mind.... I
 
 /datum/component/hattable/proc/hat_on_thing(mob/target as mob, obj/item/item as obj, mob/attacker)
 	if (src.hat)
-		return
+		item = null
+		return TRUE
```

Comment:
Won't returning true here make it impossible to attack the parent at all when it has a hat on?